src/main/scala/org/apache/spark/ml/linalg/distributed/RapidsRowMatrix.scala

```diff
@@ -91,14 +91,21 @@ class RapidsRowMatrix(
       val dense_U = DenseMatrix.zeros(n, n)
 
       val dense_S = DenseMatrix.zeros(1, n)
+      var svdDriver: Array[(DenseMatrix, DenseMatrix)] = null
       try {
-        // this is done on driver, so no task resources here, assign 0 manually.
-        RAPIDSML.calSVD(n, Cov, dense_U, dense_S, 0)
+        val rdd_tmp = rows.context.parallelize(Seq(0),1)
+        val svdRdd = rdd_tmp.mapPartitions( p => {
+          RAPIDSML.calSVD(n, Cov, dense_U, dense_S, 0)
```

Comment:
The map function needs to serialize the variables it needs and send them to the executor. So you may need to consider if you need to initialize them (like dense_U, dense_S) in the executor directly.

Reply:
Thx, optimized.

Reply:
Looks like you have specified the `device_id = 0` here, which may cause some issues when the worker has 2 GPUS. and the task is scheduled to GPU device=1.

Reply:
nice catch. Fixed.